Bird camp

alllekso opened the pull request:
tell me about bugs in my code

alllekso committed:
- add new page: bird_camp
- add example how to use remain from division by zero
Task:
Modify content to Nice

KateKrav committed:
- removed google fonts;
 - split the form and page into 2 files;
  - fixed clicking on the login in the JS file.

KateKrav committed:
-added style to the title, form and img;
-img added a hover;
- Attempt to make a sound when pointing to img (fail)

KateKrav committed:
-img added a hover (shadow);

alllekso committed:
- add switching berween img

KateKrav committed:
mp3

KateKrav committed:
mp3

Commit:
- add white side music on mousemove
- have bug: white and black music playing in the same time. Need fix to switch between music

Commit:
- add track pause when switching music

KateKrav committed:
-text, appearance, image, footer

Commit:
- style fix for "Background"

KateKrav committed:
-footer, copyright, positioning

KateKrav committed:
-footer with background and text, flex layout

KateKrav committed:
-adding favicon

KateKrav committed:
-Create a new page; social networks page. -The foundation has been laid in the creation, navigation and the first blocks. -IMG, tips, logo

KateKrav committed:
adding block contacts

KateKrav committed:
-block contacts

alllekso committed:
- soft refactor

alllekso committed:
- started redesign login page:
add sign up methods
modify login methods
not completed

KateKrav committed:
-Styling text, adding capital letters, indents and Setting the indentation of the first line of a block of text.
-Footer img

KateKrav committed:
-///

KateKrav committed:
-style first page/ background-ing, button-help,footer

KateKrav committed:
-img

KateKrav committed:
-button

KateKrav committed:
-button color

KateKrav committed:
-adding hover img avatar

KateKrav committed:
-replacement img png with svg

KateKrav committed:
-container "friends"

KateKrav committed:
-container "friends" finish

KateKrav committed:
- starting publication content

KateKrav committed:
- finished publication content

KateKrav committed:
- svg

alllekso committed:
- added "data_base" JS file
- fixed Login
- completed "next_step" manager
- hide "sign_up" button on start Login/Sign up
- added "get_selector_value" method
- added "fields_displaying_manager"

alllekso committed:
joined "login_form" & "sign_up_form" in "auth_form"

alllekso committed:
- refactoring for new architecture
- sign up completed 90%

KateKrav committed:
-changing the background, refactoring the sing up button

KateKrav committed:
-refactoring the support_form and support_form_button

KateKrav committed:
Merge remote-tracking branch 'origin/bird_camp' into bird_camp

# Conflicts:
#	index.html
#	scripts/next_step.js

KateKrav committed:
-refactoring

alllekso committed:
- fix after rename and refactoring
- added "wraps" file to store general wrap functions

alllekso committed:
- login & sign up ready. v 1.0

alllekso committed:
- login & sign up, positive cases

KateKrav committed:
-refactoring footer and button support

KateKrav committed:
-refactoring code

KateKrav committed:
-refactoring margin and padding form group

KateKrav committed:
-add margin "input-form"

junemolison commented on Black_bird-beak/black_bird_beak_style.css:
```diff
@@ -0,0 +1,276 @@
+}
+
+.navbar {
+   display:flex;
```
Нужно поддерживать консистенцию в пробелах и прочих знаках.

junemolison commented on Black_bird-beak/black_bird_beak_style.css:
```diff
@@ -0,0 +1,276 @@
+    justify-content: space-between;
+}
+
+.brand{
```
Пиксели это конечно хорошо, но это не скейлиться вообще на других размерах.
Вот тут можно почитать: 
https://medium.com/code-better/css-units-for-font-size-px-em-rem-79f7e592bb97
https://css-tricks.com/rem-global-em-local/

junemolison commented on scripts/login_page.js:
```diff
@@ -0,0 +1,194 @@
+    name_validation();
+}
+
+function name_validation() {
```
Перепиши так, чтобы функция всегда что-то возвращала, если прошло валидацию, значит возвращает тру, если нет то фолс

junemolison commented on scripts/login_page.js:
```diff
@@ -0,0 +1,194 @@
+    confirm_password = get_value("id", "confirm_pass");
+    if (
+        name_validation() === true
+        && password_validation(password) === true
```
Тебе не нужны проверки на тру, если ты перепишешь все свои валидации так, чтобы они возвращали всегда значение, соответсветнно name_validation() будет возвращать тру или фолс и в любом случае это будет влиять на выполнение проверки. То есть вместо if(true === true) ты будешь делать if(true), что одно и тоже, но меньше кода и более выразительно выглядит

junemolison commented on scripts/login_page.js:
```diff
@@ -0,0 +1,194 @@
+    if (password === "") {
+        notifications_field.innerHTML = "Pass field is required!";
+        notifications_field.className = 'error_message'
+    } else if (password.length < 3) {
```
В данном случае лучше юзвать свич кейс. Вообще вот когда и при каких условиях лучше использовать
Use switch instead of if when:

- You are comparing multiple possible conditions of an expression and the expression itself is non-trivial.
- You have multiple values that may require the same code.
- You have some values that will require essentially all of another value's execution, plus only a few statements.

Use if instead of switch when:

- You want to test for the truthiness of an expression.
- You only have a single affirmative test.
- You need to evaluate different expressions for each branch

junemolison commented on scripts/wraps.js:
```diff
@@ -0,0 +1,46 @@
+function get_element(type, selector) {
+    let element;
+    switch (type) {
```
в каждом свитч кейсе должен быть default:

Commit:
- integrated Redux.
need fix

Commit:
- redux v 0.0.1

alllekso committed:
Revert "- integrated Redux. need fix"

                                                                                    This reverts commit d2e51241

alllekso committed:
Revert "- redux v 0.0.1"

This reverts commit 1b392cc9

alllekso committed:
pull request fix:
add "Default" to Switch/case

alllekso committed:
pull request fix:
replaced condition with boolean variable on boolean assert
v. 0.9.1. - need some refactor

alllekso committed:
- changed "wraps" functions
- login bugfix
- refactor

alllekso committed:
- fixed sign up

KateKrav committed:
pull request css fix:-correction of the arrangement of spaces and other characters; -change font size from px to em.

KateKrav committed:
Merge remote-tracking branch 'origin/bird_camp' into bird_camp

KateKrav committed:
-add content publication; -add svg icons; -add video-file; -css style publication.

KateKrav committed:
-alignment 'div'; -start second sidebar; - add svg/img
